lib/auth/auth_with_roles.go

```diff
@@ -3160,6 +3157,55 @@ func (a *ServerWithRoles) DeleteAllAppSessions(ctx context.Context) error {
 	return nil
 }
 
+// DeleteUserAppSessions delete all user’s application sessions.
+func (a *ServerWithRoles) DeleteUserAppSessions(ctx context.Context, req *proto.DeleteUserAppSessionsRequest) error {
+	// First, check if the current user can delete the request user sessions.
+	if err := a.checkPermissionToDeleteUserWebSession(req.Username); err != nil {
+		return trace.Wrap(err)
+	}
+
+	sessions, err := a.authServer.GetAppSessions(ctx)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	for _, session := range sessions {
+		// The check is done against each session to avoid comparing user input
+		// with resources that the user doesn’t have access to.
+		if err := a.checkPermissionToDeleteUserWebSession(session.GetUser()); err != nil {
```

Comment:
Isn't this check redundant if we're only retrieving sessions for a particular user?